### PR TITLE
Raise the DefaultBatchMaxSize to 1000 for logs/events pipelines

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Raise the max batch payload size to `1000` elements. This is the maximum size according to [our docs](https://docs.datadoghq.com/api/latest/logs/#send-logs). 

This effectively raises the maximum throughput of logs/events (with an assumed 50ms of HTTP latency) from `~1.7 MB/sec` to `~9 MB/Sec` (tested with 1KB log samples). This has no impact on lower throughput scenarios. 

CPU usage raises proportionally to the extra log throughput, but did not exceed `900` mcores in my benchmarks. 

This will impact any pipeline using the default configs (including `epforwarder`). 

### Describe how to test/QA your changes

Send a high rate of logs to the intake. Spot check that none are missing. 

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by sgnn7 — APPROVED** [review]
@gh123man LGTM though it may be worthy of a release note

**Comment by olivielpeau** [comment]
I think this changes the default for all components sending to an event platform intake, not only logs, so let's update the PR description accordingly, communicate this changes to the affected teams and add them for QA. Also, does this limit of `1000` items apply consistently for all event platform intakes?

File: releasenotes/notes/raise_max_batch_size-c8fd6a98468ae675.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Logs and events max batch size raised from `100` to `1000` elements. Improves
```
Comment (ruthnaebeck) [inline releasenotes/notes/raise_max_batch_size-c8fd6a98468ae675.yaml:11]:
```suggestion
    Raised the max batch size of logs and events from `100` to `1000` elements. Improves
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]

**Review by olivielpeau — APPROVED** [review]

**Review by xcoderzach — APPROVED** [review]

**Review by ruthnaebeck — APPROVED** [review]
👍  for docs

**Pull request merged**